src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -177,4 +172,16 @@ private function validateKey($key)
 
         return $key;
     }
+
+    private function createGenerator(array $keys)
```

Comment:
By doing this, I am getting key validation before returning a generator. Otherwise, you will not get invalid key exception until you call the first iteration